tests/RobotFramework/tests/tedge_watchdog/tedge_watchdog.robot

```diff
@@ -0,0 +1,135 @@
+*** Settings ***
+Documentation
+...                 Test suite for tedge-watchdog service.
+...
+...                 Suite needs to check behaviour of tedge-watchdog when services respond in various ways in response
+...                 to watchdog-sent health check commands. Watchdog can check only a statically enumerated set of tedge
+...                 services, so we need to take some tedge service and override its default health check responses to
+...                 test the watchdog.
+...
+...                 Reference: docs/src/operate/monitoring/systemd-watchdog.md
+
+Resource            ../resources/common.resource
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Suite Teardown      Custom Teardown
+
+Test Tags           adapter:docker    theme:tedge-watchdog
+
+
+*** Variables ***
+# Interval at which we send notifications/check health of the service. Use smaller value for test to complete faster but
+# increase if flaky or debugging.
+${WATCHDOG_SEC}     60
+${SERVICE_NAME}     tedge-agent
+
+
+*** Test Cases ***
+# NOTE: This test succeeds, but not for the reason we expect - instead of simply not sending a notification, upon
+# reciving a status:down message tedge-watchdog exits for unknown reason, so it can't send the notification and the
+# monitored service is restarted by systemd, along with tedge-watchdog itself.
+# This is a result of a specific configuration of these services' unit files, that we're not checking or validating
+# anywhere, so we need to define this behaviour more precisely by providing additional tests.
+Watchdog doesn't send notification if service is down which leads to service restart
+    Set Service health check response    response=0
+    Restart Service    ${SERVICE_NAME}
+    Restart Service    tedge-watchdog
+
+    # wait for service to start
+    ${pid} =    Service Should Be Running    ${SERVICE_NAME}
+
+    # send status up
+    ${ts} =    Get Unix Timestamp
+
+    # wait for a health check command from watchdog
+    Should Have MQTT Messages    te/device/main/service/${SERVICE_NAME}/cmd/health/check
+
+    Sleep    ${WATCHDOG_SEC}s
+
+    # Verify that service got restarted by systemd and it's now healthy
+    ${pid1} =    Service Should Be Running    ${SERVICE_NAME}
+    Should Not Be Equal    ${pid}    ${pid1}
+
+Watchdog doesn't fail on unexpected time format
+    Set Service health check response    response=1
```

Comment:
```suggestion
    Set Service health check response    response=0
```
When you're simulating the health responses on behalf of the service from this test, the service should have be configured not to respond to the health checks by itself, right?

Reply:
You're right, as we're sending messages from inside the test, the service should not respond.